### PR TITLE
Remove Backend.glob()

### PR DESCRIPTION
**Pull request by frankenjoe**
Relates to https://github.com/audeering/audbackend/issues/46

This removes `Backend.glob()`.

### CODE REVIEW

File: tests/test_artifactory.py [hunk 1]
```diff
@@ -17,7 +17,3 @@ def test_exists(no_artifactory_access_rights):
     )
     version = '1.0.0'
     assert not BACKEND.exists(remote_file, version)
-
-
-def test_glob(no_artifactory_access_rights):
```
[the remark this answers is not in the document]
Comment (hagenw) [reply tests/test_artifactory.py:22]: This test disables temporarly the Artifactory username and API key, to make sure that `Backend.glob()` does not fail, but return an empty list. The same should be true for `Backend.ls()`, so yes, feel free to update to use `ls()`.

[the remark this answers is not in the document]
Comment (frankenjoe) [reply tests/test_artifactory.py:22]: I see. I re-added the test using `ls()`.

**Comment by frankenjoe** [comment]
I reverted the changes to the docstring example, I think it's cleaner if we do that in a separate PR.

**Comment by hagenw** [comment]
Looks good, but let's first merge https://github.com/audeering/audbackend/pull/81 and rebase here to enable running tests in the CI.

**Comment by frankenjoe** [comment]
> Looks good, but let's first merge #81 and rebase here to enable running tests in the CI.

Good catch, there was already a flake error :)

**Pull request merged**